cmd/util/cmd/epochs/cmd/recover_test.go

```diff
@@ -0,0 +1,62 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/onflow/flow-go/cmd/util/cmd/common"
+	"github.com/onflow/flow-go/model/flow"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/cmd/bootstrap/utils"
+	"github.com/onflow/flow-go/state/protocol/inmem"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// TestRecoverEpochHappyPath ensures recover epoch transaction arguments are generated as expected.
+func TestRecoverEpochHappyPath(t *testing.T) {
+	// tests that given the root snapshot, the command
+	// writes the expected arguments to stdout.
+	utils.RunWithSporkBootstrapDir(t, func(bootDir, partnerDir, partnerWeights, internalPrivDir, configPath string) {
+		internalNodes, err := common.ReadFullInternalNodeInfos(log, internalPrivDir, configPath)
+		require.NoError(t, err)
+		partnerNodes, err := common.ReadFullPartnerNodeInfos(log, partnerWeights, partnerDir)
+		require.NoError(t, err)
+
+		allNodeIds := make(flow.IdentityList, 0)
+		for _, node := range internalNodes {
+			allNodeIds = append(allNodeIds, node.Identity())
+		}
+		for _, node := range partnerNodes {
+			allNodeIds = append(allNodeIds, node.Identity())
+		}
+
+		// create a root snapshot
+		rootSnapshot := unittest.RootSnapshotFixture(allNodeIds)
+
+		snapshotFn := func() *inmem.Snapshot { return rootSnapshot }
+
+		// run command with overwritten stdout
+		stdout := bytes.NewBuffer(nil)
+		generateRecoverEpochTxArgsCmd.SetOut(stdout)
+
+		flagInternalNodePrivInfoDir = internalPrivDir
+		flagNodeConfigJson = configPath
+		flagCollectionClusters = 2
+		flagNumViewsInEpoch = 4000
+		flagNumViewsInStakingAuction = 100
+		flagEpochCounter = 2
+
+		generateRecoverEpochTxArgs(snapshotFn)(generateRecoverEpochTxArgsCmd, nil)
+
+		// read output from stdout
+		var outputTxArgs []interface{}
+		err = json.NewDecoder(stdout).Decode(&outputTxArgs)
+		require.NoError(t, err)
+		// compare to expected values
+		expectedArgs := extractRecoverEpochArgs(rootSnapshot)
+		unittest.VerifyCdcArguments(t, expectedArgs[:len(expectedArgs)-1], outputTxArgs[:len(expectedArgs)-1])
+		// @TODO validate cadence values for generated cluster assignments and clusters
```

Comment:
As part of validating this overall feature we should ideally create an integration test that deliberately triggers EFM, then executes a recovery. 

As an interim step, we can create an integration test which runs this utility, then submits the `recover_epoch` transaction, and validates that the transaction is accepted by the smart contract. We can probably do this relatively soon (once a draft of the smart contract changes are ready) to partially validate the flow.

Reply:
Added integration tests skeleton https://github.com/onflow/flow-go/pull/5576/commits/6070f4e108b7b1fbfe2d70fb55ee9835fa32467c . The full integration tests requires it own issue because it's more involved than we originally thought https://github.com/onflow/flow-go/issues/5639